ddtrace/_trace/span.py

```diff
@@ -484,9 +489,40 @@ def get_metric(self, key: _TagNameType) -> Optional[NumericType]:
         """Return the given metric or None if it doesn't exist."""
         return self._metrics.get(key)
 
+    def _validate_attribute_value(self, value):
```

Comment:
nit: Do we need this level of validation? Isn't the type checking in the method signature enough. If this is a hard requirement then this change looks fine to me.